modules/80_optimization/nash/postsolve.gms

```diff
@@ -283,38 +280,36 @@ loop(trade$(NOT tradeSe(trade)),
 
 *** critertion "infes": and are all solutions optimal?
 loop(regi,
- if((p80_repy(regi,'modelstat') ne 2) and (p80_repy(regi,'modelstat') ne 7),
-     s80_bool = 0;
-     p80_messageShow("infes") = YES;
-  );
-
+     if((p80_repy(regi,'modelstat') ne 2) and (p80_repy(regi,'modelstat') ne 7),
+         s80_bool = 0;
+         p80_messageShow("infes") = YES;
+     );
 *** critertion "nonopt": The next lines are a workaround for the status 7
 *** problem. If the objective value does not differ too much from the last known
 *** optimal solution, accept this solution as if it were optimal. 
-  if (1 le iteration.val,
-    !! no last iteration if this is the first; NA value in p80_repyLastOptim is
-    !! sticky, so test this separately
-    if (   p80_repy(regi,'modelstat') eq 7
+     p80_convNashObjVal_iter(iteration,regi) = p80_repy(regi,'objval') - p80_repyLastOptim(regi,'objval');
+     if (1 le iteration.val,
+        !! no last iteration if this is the first; NA value in p80_repyLastOptim is
+        !! sticky, so test this separately
+        if ( p80_repy(regi,'modelstat') eq 7
            !! The 1E-4 are quite arbitrary. One should do more research on how
-	   !! the solution differs over iteration when status 7 occurs. 
-       AND p80_repy(regi,'objval') - p80_repyLastOptim(regi,'objval') lt - 1e-4,
-      s80_bool = 0;
-      p80_messageShow("nonopt") = YES;     
-      display "Not all regions were status 2 in the last iteration. The deviation of the objective function from the last optimal solution is too large to be accepted:";
-      s80_dummy
-      = p80_repy(regi,'objval')
-      - p80_repyLastOptim(regi,'objval');
-      display s80_dummy;
-    );
-  );
-);
+           !! the solution differs over iteration when status 7 occurs. 
+           AND p80_convNashObjVal_iter(iteration,regi) lt - 1e-4,
```

Comment:
shouldn't this be 
```
AND abs(p80_convNashObjVal_iter(iteration,regi)) lt 1e-4,
```

Reply:
no, it was a mistake we introduced with the refactoring. We want to only accept `7` if the obj. value is not too much smaler than the obj. value from last iteration. So, the direction matters!

Reply:
Interesting, I find it odd that if the current iteration has a very different, but higher, value for `p80_repy(regi,'objval')` versus the previous iteration, we would accept the model to be stable enough to accept a non optimal solution.

Reply:
Right, but this is how it is implemented on the main develop branch right now, isn't it? If this should be changed, now would be a good time.

Reply:
Yes, this is the way it is now.
Maybe create an issue about this and assign probably Marian so he can give his opinion about changing this?